src/components/plots/barPlot.ts

```diff
@@ -151,6 +166,114 @@ export module Plot {
     }
 
     /**
+     * Retrieves the closest PlotData to queryPoint.
+     *
+     * Bars containing the queryPoint are considered closest. If queryPoint lies outside
+     * of all bars, we return the closest in the dominant direction (x for horizontal
+     * charts, y for vertical) and break ties
+     *
+     * @param {Point} queryPoint The point to which dataset points should be compared
+     *
+     * @returns {PlotData} The PlotData closest to queryPoint
+     */
+    public getClosestPlotData(queryPoint: Point): PlotData {
+      var keys = this.datasetOrder();
+      var chartXExtent = { min: 0, max: this.width() };
+      var chartYExtent = { min: 0, max: this.height() };
```

Comment:
plots have an idea of a BBox, why can't we just query against the BBox rather than hold these extents here?